File: 0001-0500/0268_missingNumber.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        target = n * (n+1) // 2
        for i in nums:
            target -= i
        return target

# Alternative solution

class Solution1:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        target = n * (n+1) // 2
        return target - sum(nums)

# Alternative solution

class Solution:
    def missingNumber(self, nums):
        nums.append(-1)
        i = 0
        while i < len(nums):
            j = nums[i]
            if nums[i] != nums[j] and nums[i] != -1:
                nums[i], nums[j] = nums[j], nums[i]
            else:
                i += 1
        for i, num in enumerate(nums):
            if num == -1:
                return i

File: 0001-0500/0268_missingNumber.java
class Solution {
    public int missingNumber(int[] nums) {
        int i = 0;
        int temp;

        while (i < nums.length) {
            while (nums[i] != nums.length && i != nums[i]) {
                temp = nums[i];
                nums[i] = nums[nums[i]];
                nums[temp] = temp;
            }
            i++;
        }

        for (int j = 0; j < nums.length; j++) {
            if (nums[j] != j) {
                return j;
            }
        }

        return nums.length;
    }
}
